src/TheLow.sol

```diff
@@ -1,49 +1,163 @@
-// SPDX-License-Identifier: UNLICENSED
-pragma solidity ^0.8.13;
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.13;
 
 import {ERC721} from "solmate/tokens/ERC721.sol";
+import {Owned} from "solmate/auth/Owned.sol";
+import "./utils.sol";
 
-contract TheLow is ERC721 {
+/// @notice
+contract TheLow is ERC721, Owned {
     //
+    event SupplyUpdated(uint8 indexed newSupply);
+    // event MetadataUpdated(string uri);
+    // event MetadataFrozen();
 
-    string[] internal _tierURIs;
-    mapping(uint256 => string) _tokenURIs;
+    uint8 public constant MAX_SUPPLY = 222;
+    uint8 public totalSupply = 222;
+
+    struct Tier {
+        string name;
+        string rarity;
+        string image_cid;
+        string animation_cid;
+        string animation_hash;
+        uint8 portion;  // How many tokenIds, out of 222, of this type will be created
+    }
+
+    struct RandBytes {
+        bytes32 data;
+        uint8 index;
+    }
+   Tier[6] internal _tierInfo;
+   uint8[MAX_SUPPLY+1] internal _tokenTiers;  // TokenIds are 1-indexed
```

Comment:
This seems a little non-idiomatic to set the size in this way. Could think through storage layout implications too, but this is probably fine.